parser/expression_test.go

```diff
@@ -6293,24 +6284,42 @@ func TestParseStringTemplate(t *testing.T) {
 			}
 		}
 
+		require.NoError(t, err)
+	})
+
+	t.Run("invalid, missing paren", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, errs := testParseExpression(`
+			"\(add"
+		`)
+
+		var err error
+		if len(errs) > 0 {
+			err = Error{
+				Errors: errs,
+			}
+		}
```

Comment:
Instead of creating a new error, we can simply check whether the `errs` are empty. i.e:
```suggestion
		require.NotEmpty(t, errs)
```

Same for other similar places as well.